substrate/frame/revive/src/vm/evm.rs

```diff
@@ -110,32 +148,148 @@ pub fn call<'a, E: Ext>(bytecode: Bytecode, ext: &'a mut E, inputs: EVMInputs) -
 	let table = instruction_table::<'a, E>();
 	let result = run(&mut interpreter, &table);
 
-	if result.is_error() {
-		Err(Error::<E::T>::ContractTrapped.into())
-	} else {
-		Ok(ExecReturnValue {
-			flags: if result.is_revert() { ReturnFlags::REVERT } else { ReturnFlags::empty() },
-			data: result.output.to_vec(),
+	instruction_result_into_exec_error::<E>(result.result)
+		.map(Err)
+		.unwrap_or_else(|| {
+			Ok(ExecReturnValue {
+				flags: if result.is_revert() { ReturnFlags::REVERT } else { ReturnFlags::empty() },
+				data: result.output.to_vec(),
+			})
 		})
-	}
 }
 
-/// Runs the EVM interpreter
-fn run<WIRE: InterpreterTypes>(
-	interpreter: &mut Interpreter<WIRE>,
-	table: &revm::interpreter::InstructionTable<WIRE, DummyHost>,
+/// Runs the EVM interpreter until it returns an action.
```

Comment:
```suggestion
/// Runs the EVM interpreter
```

or and handle actions if you prefer